src/recordlinker/linking/link.py

```diff
@@ -82,16 +82,16 @@ def add_person_resource(
     return bundle
 
 
-def compare(record: models.PIIRecord, patient: models.Patient, linkage_pass: dict) -> bool:
+def compare(record: models.PIIRecord, patient: models.Patient, algorithm_pass: models.AlgorithmPass) -> bool:
     """
     Compare the incoming record to the linked patient
     """
     # all the functions used for comparison
-    funcs: dict[models.Feature, matchers.FEATURE_COMPARE_FUNC] = linkage_pass["funcs"]
+    funcs: dict[models.Feature, matchers.FEATURE_COMPARE_FUNC] = utils.bind_functions(algorithm_pass.evaluators)
```

Comment:
I was thinking we'd do this a bit earlier.  I think likely when we pull it out of the database, but I haven't really thought that through, so maybe just doing it at the top of link is safest for now.  Anyway, if we have 20 patient records come back, 2 passes and 2 evaluation features to compare, we call this method 80 times.  If we do the binding earlier in the stack, we can reduce the bind_functions overhead.

Reply:
I was thinking about doing this but the `utils.bind_functions` takes in a `dict` and modifies the function `str` to a `callable` I put it here to avoid using a dict and keep `AlgorithmPass` everywhere else.

Reply:
Ah, that's a great point, if we modify it it's really no longer a AlgorithmPass object to pass around.  I think there is room here for a future optimization with the Algorithm schema class I'm working on, but no need to hold this up for that.  Would you mind just adding a TODO comment here explaining the opportunity for an optimization if we can bind earlier in the stack, then I think we'll be good.